src/model.coffee

```diff
@@ -169,6 +169,24 @@ class Model extends Backbone.Model
       )
       .promise(options.returnValues.jqXhr)
 
+  destroy: (options = {}) ->
+    cleanUpAssociatedReferences = =>
+      for collectionName, collection of @storageManager.collections
+        for reference, modelName of collection.modelKlass.associations
+          associationKey = collection.modelKlass.associationDetails(reference).key
+          if _.isArray(modelName) && _.find(modelName, (m) => inflection.singularize(m) == @className())
+            collection.storage.each (model) =>
+              model.set(associationKey, _.without(model.get(associationKey), @id))
+          else if !_.isArray(modelName) && inflection.singularize(modelName) == @className()
+            collection.storage.each (model) =>
+              model.unset(associationKey)
+
+    if(options.wait)
+      super(options).then(cleanUpAssociatedReferences)
+    else
+      cleanUpAssociatedReferences()
+      super(options)
```

Comment:
Instead of this loop, we can rely on the `destroy` event that the model triggers on `destroy` (http://backbonejs.org/docs/backbone.html#section-89). Given that we can convert this entirely into the following

`this.listenTo this, 'destroy', cleanUpAssociatedReferences`

It looks funny, by using `listenTo` allows us to rely on the automatic clean up when views are closed (http://marionettejs.com/docs/v1.8.8/marionette.view.html#view-close)